From a400ac3d83023a66a356d056899d6b380cb30473 Mon Sep 17 00:00:00 2001
From: Chase Maupin <Chase.Maupin@ti.com>
Date: Wed, 7 Mar 2012 10:51:43 -0600
Subject: [PATCH] Makefile: Update makefile to work with OE

* Updated the makefile to with OE
* Use the kernel install target for installing the module

Upstream-Status: Pending
    * will be in next release

Signed-off-by: Chase Maupin <Chase.Maupin@ti.com>
---
 Makefile |   45 ++++++++++++++++++++++++++++++++++++++++-----
 1 files changed, 40 insertions(+), 5 deletions(-)

diff --git a/Makefile b/Makefile
index ebbcd11..b17d33e 100755
--- a/Makefile
+++ b/Makefile
@@ -7,15 +7,50 @@ else
   EXTRA_CFLAGS += -O2
 endif

+-include ../../../Rules.make
+
+# If KERNEL_DIR is not set then use the default in Rules.make
+KERNEL_DIR ?= ${LINUXKERNEL_INSTALL_DIR}
+DEST_DIR ?= ${DESTDIR}
+
+PLATFORM ?= "unknown"
+MACHINE_NAME ?= "unknown"
+
+# Use the PLATFORM value from the Rules.make if it was sourced
+ifeq ($(PLATFORM), am335x-evm)
+    MACHINE_NAME := "am335x"
+endif
+ifeq ($(PLATFORM), am180x-evm)
+    MACHINE_NAME := "am1808"
+endif
+ifeq ($(PLATFORM), da850-omapl138-evm)
+    MACHINE_NAME := "am1808"
+endif
+ifeq ($(PLATFORM), am37x-evm)
+    MACHINE_NAME := "omap3evm"
+endif
+
+# If CROSS_COMPILE is not set by Rules.make then set a sane default
+CROSS_COMPILE ?= arm-arago-linux-gnueabi-
+export CROSS_COMPILE
+
+# set the INSTALL_MOD_DIR so that the executables won't be placed in extra
+INSTALL_MOD_DIR = kernel/drivers/bt_enable
+export INSTALL_MOD_DIR
+
 obj-m := gpio_en.o

+MAKE_ENV = ARCH=arm
+
 PWD := $(shell pwd)
 all:
-	pwd
-	@echo EXTRA_CFLAGS = $(EXTRA_CFLAGS)
-	$(MAKE) CROSS_COMPILE=$(CROSS_COMPILE) ARCH=$(ARCH) EXTRA_CFLAGS="$(EXTRA_CFLAGS)" -C $(KERNEL_DIR) M=$(PWD) modules
+	@cp -f gpio_en_${MACHINE_NAME}.c gpio_en.c
+	$(MAKE) EXTRA_CFLAGS="$(EXTRA_CFLAGS)" -C $(KERNEL_DIR) $(MAKE_ENV) \
+    M=$(PWD) modules
+
 install:
-	install -d ${DEST_DIR}${BASE_LIB_DIR}/modules/${KRNL_VER}/kernel/drivers/bt_enable
-	install -m 0755 ./gpio_en.ko ${DEST_DIR}${BASE_LIB_DIR}/modules/${KRNL_VER}/kernel/drivers/bt_enable
+	$(MAKE) EXTRA_CFLAGS="$(EXTRA_CFLAGS)" -C $(KERNEL_DIR) $(MAKE_ENV) \
+    M=$(PWD) INSTALL_MOD_PATH="${DEST_DIR}" modules_install
+
 clean:
	rm -rf *.o *~ core .depend .*.cmd *.ko *.mod.c .tmp_versions *.symvers
--
1.7.0.4
